File: webApp/src/models/Destination.ts
export type DestinationSortField =
  | "name"
  | "country"
  | "city"
  | "category"
  | "createdAt"
  | "updatedAt";

export type SortDirection = "asc" | "desc";

export interface Destination {
  id: string;
  name: string;
  country: string;
  countryCode?: string | null;
  city: string | null;
  description: string;
  category: string;
  latitude: number | null;
  longitude: number | null;
  coverImageUrl: string | null;
  createdAt: string;
  updatedAt: string;
  dataOrigin?: "TOURVERSE_CURATED" | "EXTERNAL" | "HYBRID" | "DEVELOPMENT_SEED";
  lastVerifiedAt?: string | null;
  verificationStatus?: "VERIFIED" | "PARTIALLY_VERIFIED" | "REVIEW_REQUIRED" | "REJECTED";
  attributionSummary?: string | null;
  mapAvailable?: boolean;
  googlePlaceId?: string | null;
}

export interface PagedDestinationResponse {
  items: Destination[];
  page: number;
  size: number;
  totalItems: number;
  totalPages: number;
}

export interface DestinationQuery {
  search?: string;
  country?: string;
  countryCode?: string;
  city?: string;
  category?: string;
  page: number;
  size: number;
  sortBy: DestinationSortField;
  sortDirection: SortDirection;
}

export interface ApiMessage {
  status: string;
  message: string;
}

export interface DestinationCountry {
  code: string;
  name: string;
  destinationCount: number;
}

export interface DestinationCountriesResponse {
  countries: DestinationCountry[];
}

File: frontend/androidApp/app/src/main/kotlin/com/tourverse/data/model/Destination.kt
package com.tourverse.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Destination(
    val id: String,
    val name: String,
    val country: String,
    val city: String? = null,
    val description: String,
    val category: String,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val coverImageUrl: String? = null,
    val createdAt: String,
    val updatedAt: String,
    val countryCode: String? = null,
    val dataOrigin: String = "TOURVERSE_CURATED",
    val lastVerifiedAt: String? = null,
    val verificationStatus: String = "VERIFIED",
    val attributionSummary: String? = null,
    val mapAvailable: Boolean = latitude != null && longitude != null,
    val googlePlaceId: String? = null
) {
    val displayLocation: String
        get() = listOfNotNull(city?.trim()?.takeIf(String::isNotEmpty), country.trim())
            .joinToString(", ")
}

@Serializable
data class DestinationCountry(
    val code: String,
    val name: String,
    val destinationCount: Long
)

@Serializable
data class DestinationCountriesResponse(val countries: List<DestinationCountry>)
